src/RequestHandle.cpp

```diff
@@ -401,6 +470,19 @@ auto RequestHandle::onComplete() -> void {
     m_on_complete_handler(std::move(request));
 }
 
+
+auto RequestHandle::getBytesLeftToWrite() -> size_t
```

Comment:
getDownloadBytesRemaining?  getDownloadBytesLeft?  .. getRemainingDownloadBytes? I know curl uses "write" but its a confusing term in my opinion, is that the number of bytes to write in the REQUEST or "read" in the RESPONSE?

Reply:
Went with getRemainingDownloadBytesOptional, since this now returns an optional as part of bounds checking.